darts/timeseries.py

```diff
@@ -1612,12 +1663,30 @@ def pd_dataframe(self, copy=True, suppress_warnings=False) -> pd.DataFrame:
         ----------
         copy
             Whether to return a copy of the dataframe. Leave it to True unless you know what you are doing.
+        backend
+            The backend to which to export the `TimeSeries`. See the `narwhals documentation
+            <https://narwhals-dev.github.io/narwhals/api-reference/narwhals/#narwhals.from_dict>`_ for all supported
+            backends.
+        time_as_index
+            Whether to set the time index as the index of the dataframe or in the left-most column.
+        suppress_warnings
+            Whether to suppress the warning about transforming a stochastic TimeSeries
 
         Returns
         -------
-        pandas.DataFrame
-            The Pandas DataFrame representation of this time series
+        DataFrame
+            A DataFrame representation of this time series.
         """
+
+        if time_as_index and backend != "pandas":
+            raise_log(
+                Warning(
+                    '`time_as_index=True` is only supported with `backend="pandas"`.'
+                ),
+                logger,
+            )
```

Comment:
this would raise an error, we only want a warning

```suggestion
            logger.warning('`time_as_index=True` is only supported with `backend="pandas"`.')
            time_as_index = False
```